### PR TITLE
Improve Mesh layer description v2

### LINKED ISSUE #2908: [mesh] [feature] add opacity slider to mesh contours settings
Original commit: https://github.com/qgis/QGIS/commit/d7afb710d72ad5a65262400030419beb5b48ea16 by wonder-sk

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #2943: [mesh] [feature] allow to choose different vector and scalar dataset
Original commit: https://github.com/qgis/QGIS/commit/f0ede773c20b8f3720e75adbe95cc49d01d7d3e1 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by rosaguilar**
### Description

Goal:
To adjust the text to include suggestions and subsections (supersedes #3565)

Ticket: fix #2908 fix #2943  #2967  #2752

- [ x] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 1]
```diff
@@ -142,9 +142,96 @@ The :guilabel:`Source` tab displays basic information about the selected mesh, i
...
+
+
+.. figure:: img/styling_panel.png
+   :align: center
+
+   Mesh Layer Style
+
+In the tab |rendering_mesh|,  QGIS offers two possibilities to display the grid:
+
+*	Native mesh rendering that shows quadrants
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
*	``Native mesh rendering`` that shows quadrants
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 2]
```diff
@@ -142,9 +142,96 @@ The :guilabel:`Source` tab displays basic information about the selected mesh, i
...
+*	Triangular mesh rendering that display triangles
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
*	``Triangular Mesh Rendering`` that display triangles
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 3]
```diff
@@ -142,9 +142,96 @@ The :guilabel:`Source` tab displays basic information about the selected mesh, i
...
+
+The line width and color can be changed in this dialog or deactivate both grid rendering.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
In this dialog the line width and color can be changed, and the rendering options can be activated / deactivated.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 4]
```diff
@@ -142,9 +142,96 @@ The :guilabel:`Source` tab displays basic information about the selected mesh, i
...
+
+The tab |tbutton| presents the following items:
+
+* groups available in the mesh dataset
+* dataset in the selected group(s), for example, if the layer has a temporal dimension
+* metadata if available
+* blending mode available for the selected dataset, for example: normal, lighten,multiply,…(see blending modes :ref:`blend-modes`)
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
* blending mode available for the selected dataset, for example: normal, lighten, multiply,…(see blending modes :ref:`blend-modes`)
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 5]
```diff
@@ -142,9 +142,96 @@ The :guilabel:`Source` tab displays basic information about the selected mesh, i
...
+.. figure:: img/contours_style.png
+   :align: center
+
+   Styling Contours in a Mesh Layer
+
+Use the slide bar or combo box to set the opacity of the current group.
+
+Use :guilabel:`Load` to adjust the min and max value of the current group.
+
+The :guilabel:`Interpolation` list contains three option to render contours: Linear, Discrete and Exact.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
The :guilabel:`Interpolation` list contains three options to render contours: Linear, Discrete and Exact.
```

**Comment by DelazJ** [comment]
@rosaguilar there does not seem to be a difference between this pull request and #3565 (same branch with same commits so will call same comments). To address the comments in #3565, you have to edit the files in your branch and create a new commit that (if pushed online) will automatically update the PR. you don't need a new PR unless it' for another issue (_this pr should be closed_).

If you need more help on how to make changes on an existing pull request, you may give a look to our step by step guidelines (https://docs.qgis.org/testing/en/docs/documentation_guidelines/first_contribution.html#make-corrections). This one focuses on changes in Github but the idea is the same if you do your changes locally and push you branch to the web. Or just ask; i think we have enough git guru around to help you go through this.

**Comment by rosaguilar** [comment]
> @rosaguilar there does not seem to be a difference between this pull request and #3565 (same branch with same commits so will call same comments). To address the comments in #3565, you have to edit the files in your branch and create a new commit that (if pushed online) will automatically update the PR. you don't need a new PR unless it' for another issue (_this pr should be closed_).
> 
> If you need more help on how to make changes on an existing pull request, you may give a look to our step by step guidelines (https://docs.qgis.org/testing/en/docs/documentation_guidelines/first_contribution.html#make-corrections). This one focuses on changes in Github but the idea is the same if you do your changes locally and push your branch to the web. Or just ask; i think we have enough git guru around to help you go through this.

Hi, The idea was to add the subsections as you suggested and remove unused icons. Also, to request backport to LTR. Clearly, something went wrong. Sorry about that. I will back to this later.

**Comment by rosaguilar** [comment]
@DelazJ @havatv  I have finished and push my changes. Would you like to take a look?

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 6]
```diff
@@ -137,15 +136,136 @@ Source
...
+Enter values on :guilabel:`Min` and :guilabel:`Max` to filter vectors according to their magnitude.
+
+Check on the box |checkbox| :guilabel:`Display Vectors on User Grid` and specify the :guilabel:`X spacing` and the :guilabel:`Y spacing`,
+QGIS will render the vector considering the given spacing.
+
+On the Head Options :guilabel:`Head Options`, QGIS allows to set the shape of the arrow head by specifying width and length (in percentage).
+
+Vector's :guilabel:`Arrow length` can be rendered in QGIS in three different ways:
+
+* Defined by Min and Max: You should specify the minimum and maximum length for vector,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
* Defined by Min and Max: You should specify the minimum and maximum length for the vectors,
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 7]
```diff
@@ -137,15 +136,136 @@ Source
...
+  QGIS will adjust their visualization accordingly
+* Scale to magnitude: You should specify the (multiplying) factor to use
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
* Scale to magnitude: You specify the (multiplying) factor to use
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 8]
```diff
@@ -137,15 +136,136 @@ Source
...
+
+Rendering
+.........
+
+In the tab |rendering_mesh|,  QGIS offers two possibilities to display the grid:
+
+*	``Native Mesh Rendering`` that shows quadrants
+*	``Triangular Mesh Rendering`` that display triangles
+
+The line width and color can be changed in this dialog or deactivate both grid rendering.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
The line width and color can be changed in this dialog, and both the grid renderings can be turned off.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 9]
```diff
@@ -137,15 +136,136 @@ Source
...
+In the tab |general|, click on |vector_inactive| to display vectors if available.
+The map canvas will display the vectors in the selected group with default parameters.
+Click on the tab |vector_active| to change the visualization parameters for vectors, see figure below:
+
+.. figure:: img/vector_styling.png
+   :align: center
+
+   Styling Vectors in a Mesh Layer
+
+The line width can be set using the combo box or typing the value. The color list, opens the dialog :guilabel:`Change color`,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
The line width can be set using the combo box or typing the value.
The color list, opens the dialog :guilabel:`Change color`,
```

Comment (havatv) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: Shortening the line.

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 10]
```diff
@@ -137,15 +136,136 @@ Source
...
+where you can select a color to apply to vectors.
+
+Enter values on :guilabel:`Min` and :guilabel:`Max` to filter vectors according to their magnitude.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
Enter values for :guilabel:`Min` and :guilabel:`Max` to filter vectors according to their magnitude.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 11]
```diff
@@ -137,15 +136,136 @@ Source
...
+
+Check on the box |checkbox| :guilabel:`Display Vectors on User Grid` and specify the :guilabel:`X spacing` and the :guilabel:`Y spacing`,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
Check on the box |checkbox| :guilabel:`Display Vectors on User Grid` and specify
the :guilabel:`X spacing` and the :guilabel:`Y spacing`,
```

Comment (havatv) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: shorter line

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 12]
```diff
@@ -137,15 +136,136 @@ Source
...
+QGIS will render the vector considering the given spacing.
+
+On the Head Options :guilabel:`Head Options`, QGIS allows to set the shape of the arrow head by specifying width and length (in percentage).
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
On the Head Options :guilabel:`Head Options`, QGIS allows to set the shape of the arrow head
by specifying width and length (in percentage).
```

Comment (havatv) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: shorter line

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 13]
```diff
@@ -137,15 +136,136 @@ Source
...
+  QGIS will adjust their visualization accordingly
+* Scale to magnitude: You should specify the (multiplying) factor to use
+* Fixed: all the vectors are shown with the same length
+
+.. _mesh_symbology_rendering:
+
+Rendering
+.........
+
+In the tab |rendering_mesh|,  QGIS offers two possibilities to display the grid:
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]: Is there an interesting screenshot for this tab?

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 14]
```diff
@@ -160,4 +280,26 @@ TODO
...
+   :width: 1.5em
+.. |contour_inactive| image:: img/contours_inactive.png
+   :width: 2em
+.. |vector_active| image:: img/vector_active.png
+   :width: 2em
+.. |vector_inactive| image:: img/vector_inactive.png
+   :width: 1.5em
+.. |checkbox| image:: /static/common/checkbox.png
+   :width: 1.5em
+.. |slider| image:: /static/common/hslider.png
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]: this is not the slider used as far as i can see. Isn't it instead this one https://github.com/qgis/QGIS-Documentation/blob/master/static/common/slider.png, referenced at https://github.com/qgis/QGIS-Documentation/blob/master/source/substitutions.txt#L805, ie 
```suggestion
.. |slider| image:: /static/common/slider.png
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 15]
```diff
@@ -137,15 +136,136 @@ Source
...
+
+By selecting ‘Continuous’ in the classification :guilabel:`Mode`, QGIS creates classes automatically considering
+the :guilabel:`Min` and :guilabel:`Max` values. With ‘Equal interval’, you only need to select the number of classes
+using the combo box  :guilabel:`Classes`  and press the button :guilabel:`Classify`.
+
+The button |signPlus| :sup:`Add values manually` adds a value
+to the individual color table. The button |signMinus| :sup:`Remove selected row`
+deletes a value from the individual color table. Double clicking on the value column
+lets you insert a specific value. Double clicking on the color column opens the dialog
+:guilabel:`Change color`, where you can select a color to apply on that value.
```
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: @DelazJ you are right. I was not sure how to proceed. So, I left to get feedback from you. Feel free to squash it.

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 16]
```diff
@@ -137,15 +136,140 @@ Source
...
+The map canvas will display the vectors in the selected group with default parameters.
+Click on the tab |vector_active| to change the visualization parameters for vectors, see figure below:
+
+.. figure:: img/vector_styling.png
+   :align: center
+
+   Styling Vectors in a Mesh Layer
+
+The line width can be set using the combo box or typing the value.
+The color list, opens the dialog :guilabel:`Change color`,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
The color widget opens the dialog :guilabel:`Change color`,
```

**Comment by DelazJ** [comment]
@rosaguilar Looks like you removed some icons instead of 
> **Move** the other icons (**contours_active, contours_inactive, rendering** (see below), **groups_exploring** (?), **vector_active, vector_inactive**) from the relative ./img folder to the /static/common folder (where all icons are placed). Only screenshots are expected to be in the relative img folder.

You should indeed place the missing icons you are adding in the /static/common/ folder and (I also forgot the last time, sorry) reference them in the `/source/substitutions.txt` file.
Ideally, they would need to be also mentioned in the `source/docs/documentation_guidelines/substitutions.rst` so that they are displayed [here](https://docs.qgis.org/testing/en/docs/documentation_guidelines/substitutions.html#layer-properties) and prevent people trying to add them again. But this is something we can also take care of in a later step.

**Comment by rosaguilar** [comment]
> @rosaguilar Looks like you removed some icons instead of
> 
> > **Move** the other icons (**contours_active, contours_inactive, rendering** (see below), **groups_exploring** (?), **vector_active, vector_inactive**) from the relative `./img` folder to the `/static/common` folder (where all icons are placed). Only screenshots are expected to be in the relative img folder.
> 
> You should indeed place the missing icons you are adding in the /static/common/ folder and (I also forgot the last time, sorry) reference them in the `/source/substitutions.txt` file.
> Ideally, they would need to be also mentioned in the `source/docs/documentation_guidelines/substitutions.rst` so that they are displayed [here](https://docs.qgis.org/testing/en/docs/documentation_guidelines/substitutions.html#layer-properties) and prevent people trying to add them again. But this is something we can also take care of in a later step.

Hi, I followed your suggestions and did the substitutions. 
 I tried to moved, but for some reasons I am not getting them in the static/common

**Comment by DelazJ** [comment]
>  I tried to moved, but for some reasons I am not getting them in the static/common

Can you elaborate, please ? How did you try the move?

**Comment by rosaguilar** [comment]
> > I tried to moved, but for some reasons I am not getting them in the static/common
> 
> Can you elaborate, please ? How did you try the move?

I commented the .gitignore to track my local additions (i added the small icons to common/static)

**Comment by DelazJ** [comment]
Your files are placed in source/static/common instead of static/common

**Comment by rosaguilar** [comment]
> Your files are placed in source/static/common instead of static/common

Oh... sorry. How can I fix that?

**Comment by DelazJ** [comment]
Do you have a local repo you are using or are you doing all your changes through github? I'm particularly interested in how you made 1079647

**Comment by DelazJ** [comment]
Ah I see you upload them to the right place. Don't forget to drop the source/static folder

**Comment by rosaguilar** [comment]
> Do you have a local repo you are using or are you doing all your changes through github? I'm particularly interested in how you made [1079647](https://github.com/qgis/QGIS-Documentation/commit/1079647a0d961eb37c5e2dcce8d34e1ab86ad1f3)

As you may see, it has been a bit chaotic..sorry for that

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 17]
```diff
@@ -137,15 +136,165 @@ Source
...
+The tab |general| presents the following items:
+
+* groups available in the mesh dataset
+* dataset in the selected group(s), for example, if the layer has a temporal dimension
+* metadata if available
+* :ref:`blending mode <blend-modes>` available for the selected dataset.
+
+The slider |slider|, combo box |selectString| and |groupsexploring| buttons allow to explore another dimension of the data, if available.
+As the slider moves, the metadata is presented accordingly.
+See the figure :ref:`Mesh groups <figure_mesh_groups_>` below as an example.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
See the figure :ref:`Mesh groups <figure_mesh_groups>` below as an example.
```
(_) Not needed when using :ref:

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 18]
```diff
@@ -137,15 +136,165 @@ Source
...
+.. _mesh_symbology_contours:
+
+Contours Symbology
+..................
+
+Under :guilabel:`Groups`, click on |contour_inactive| to show contours with
+default visualization parameters.
+
+In the tab |contour_active| you can see and change the current visualization options of contours
+for the selected group, as shown in the image :ref:`Styling contours in a mesh <figure_mesh_symbology_contours_>` below:
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
for the selected group, as shown in the image :ref:`Styling contours in a mesh
<figure_mesh_symbology_contours>` below:
```

**Comment by DelazJ** [comment]
> As you may see, it has been a bit chaotic..sorry for that

That's not an issue. It can be cleaned afterwards. I'd say, the most important is to understand the logic behind and why it's failing.

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 15 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: OK. Reported at #3598

**Comment by DelazJ** [comment]
OK. Travis CI is happy now 😃 ! And the PR is in a rather good shape (the remaining points can be handled in another step)
In order to clean the history of your modifications, can you reply to my previous question:
> Do you have a local repo you are using or are you doing all your changes through github (web interface)?

I understood you were on Windows: are you also using eg the [Github Desktop](https://desktop.github.com/) software? In which case, using menus-only or git command line tools?

**Comment by rosaguilar** [comment]
> OK. Travis CI is happy now 😃 ! And the PR is in a rather good shape (the remaining points can be handled in another step)
> In order to clean the history of your modifications, can you reply to my previous question:
> 
> > Do you have a local repo you are using or are you doing all your changes through github (web interface)?
> 
> I understood you were on Windows: are you also using eg the [Github Desktop](https://desktop.github.com/) software? In which case, using menus-only or git command line tools?
I was using git command line tools basically

**Comment by rosaguilar** [comment]
> Your files are placed in source/static/common instead of static/common

Thanks!.

**Comment by rosaguilar** [comment]
> Do you have a local repo you are using or are you doing all your changes through github? I'm particularly interested in how you made [1079647](https://github.com/qgis/QGIS-Documentation/commit/1079647a0d961eb37c5e2dcce8d34e1ab86ad1f3)

Thanks to @rduivenvoorde I got a venv where I could build myself the local repo (in that way was easy for me to see the results e.g., rst labels.. etc). I used git command line tools to update changes from my local repo ... but after some comments/suggestions, commits, etc. It became messy. For example, I tried to commit suggestions that were outdated, and I was a bit confused about what I should or not to do.  I needed extra help to synchronize everything again (locally),  after commit some local changes I decided to continue editing in github and synchronizing just partially in my local repo.

**Comment by DelazJ** [comment]
Below a recipe (from a self-taught and somehow git beginner) that might work to squash all these commits  in a single one, assuming that `git remote -v` returns _origin_ for your online repository and _upstream_ for the qgis/QGIS-Documentation's

``` bash
# update your master branch with the one at qgis repo, by first fetching the upstream data
git checkout master
git fetch upstream master
# pull them in your master branch. I notice that your online master branch is derived from qgis/master but if it's not the case for the local master branch then this should work
git pull upstream master
# otherwise use the one below
git reset --hard upstream/master
# then push the changes online
git push -f origin master
```
In github.com, your master should be now inline with qgis/master, and same for the one you locally have. 
side note: This is my preferred way to update my master branch; I don't like using `git merge` to pull the upstream modifications in my branch.
Now let's apply the changes in this PR on top of the master branch
```bash
git checkout 'meshlayerwork'
git rebase master
#  here you may get two conflicts in mesh_properties file: open it in your text editor and fix them
# (you'd want to remove the HEAD part in both cases)
#  once everything is fixed
git add source/docs/user_manual/working_with_mesh/mesh_properties.rst
git rebase --continue
#  again another conflict, in the same file with again removing HEAD part
git add source/docs/user_manual/working_with_mesh/mesh_properties.rst
git rebase --continue
#  then the rest should process smoothly
```
Now you have your changes in your local branch on top of the master branch
Let's squash the about forty commits
```bash
git reset master
```
This brings the head commit same as the one of master and, on top of it you have all your changes in an unmerged state. You can then choose the files to add (I don't know whether the gitignore or the make.bat files should be commited - I let more skilled people comment that). Otherwise
```bash
git add .
git commit -m "put here your commit message"
# let's update the pull request by pushing the changes online
git push -f origin 'meshlayerwork'
```
And you should be done.
Hope that helps and hope that works.

**Comment by rosaguilar** [comment]
> Below a recipe (from a self-taught and somehow git beginner) that might work to squash all these commits in a single one, assuming that `git remote -v` returns _origin_ for your online repository and _upstream_ for the qgis/QGIS-Documentation's
> 
> ```shell
> # update your master branch with the one at qgis repo, by first fetching the upstream data
> git checkout master
> git fetch upstream master
> # pull them in your master branch. I notice that your online master branch is derived from qgis/master but if it's not the case for the local master branch then this should work
> git pull upstream master
> # otherwise use the one below
> git reset --hard upstream/master
> # then push the changes online
> git push -f origin master
> ```
> 
> In github.com, your master should be now inline with qgis/master, and same for the one you locally have.
> side note: This is my preferred way to update my master branch; I don't like using `git merge` to pull the upstream modifications in my branch.
> Now let's apply the changes in this PR on top of the master branch
> 
> ```shell
> git checkout 'meshlayerwork'
> git rebase master
> #  here you may get two conflicts in mesh_properties file: open it in your text editor and fix them
> # (you'd want to remove the HEAD part in both cases)
> #  once everything is fixed
> git add source/docs/user_manual/working_with_mesh/mesh_properties.rst
> git rebase --continue
> #  again another conflict, in the same file with again removing HEAD part
> git add source/docs/user_manual/working_with_mesh/mesh_properties.rst
> git rebase --continue
> #  then the rest should process smoothly
> ```
> 
> Now you have your changes in your local branch on top of the master branch
> Let's squash the about forty commits
> 
> ```shell
> git reset master
> ```
> 
> This brings the head commit same as the one of master and, on top of it you have all your changes in an unmerged state. You can then choose the files to add (I don't know whether the gitignore or the make.bat files should be commited - I let more skilled people comment that). Otherwise
> 
> ```shell
> git add .
> git commit -m "put here your commit message"
> # let's update the pull request by pushing the changes online
> git push -f origin 'meshlayerwork'
> ```
> 
> And you should be done.
> Hope that helps and hope that works.

@DelazJ Thanks a lot  :)

**Comment by DelazJ** [comment]
Glad it works.
Because I really would like to have this automagically backported to 3.4 (and some would say i'm a nitpicker 😄 - but do not listen to them), can you
```bash
git checkout 'meshlayerwork'
git reset --hard upstream/master
git cherry-pick 6d7b3ba
git cherry-pick 1f73b5e
git push -f origin 'meshlayerwork'
```
This will erase the two merge commits that are of master modifications and couldn't be backported without manual changes to 3.4. Thanks.

**Comment by rosaguilar** [comment]
> Glad it works.
> Because I really would like to have this automagically backported to 3.4 (and some would say i'm a nitpicker 😄 - but do not listen to them), can you
> 
> ```shell
> git checkout 'meshlayerwork'
> git reset --hard upstream/master
> git cherry-pick 6d7b3ba
> git cherry-pick 1f73b5e
> git push -f origin 'meshlayerwork'
> ```
> 
> This will erase the two merge commits that are of master modifications and couldn't be backported without manual changes to 3.4. Thanks.

Done!. Lets see if Travis is happy now.

**Comment by DelazJ** [comment]
Thanks @rosaguilar . Great work!

**Comment by rosaguilar** [comment]
> Thanks @rosaguilar . Great work!

Thanks to everyone that helped!

**Comment by saberraz** [comment]
Thanks a lot @rosaguilar from myself and my colleagues. It's been an enormous task.

**Comment by ghtmtt** [comment]
thanks @rosaguilar for this work! Nice to see new people working on the documentation. You've picked up a super huge topic for the first contribution :) Thanks also to all the people that helped with suggestions and reviews!

**Pull request merged**